src/components/app/details/triggerView/BranchRegexModal.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react'
+import { ReactComponent as GitLab } from '../../../../assets/icons/git/gitlab.svg'
+import { ReactComponent as Git } from '../../../../assets/icons/git/git.svg'
+import { ReactComponent as GitHub } from '../../../../assets/icons/git/github.svg'
+import { ReactComponent as BitBucket } from '../../../../assets/icons/git/bitbucket.svg'
+import { SourceTypeMap } from '../../../../config'
+import { ReactComponent as Close } from '../../../../assets/icons/ic-close.svg'
+import { ReactComponent as LeftIcon } from '../../../../assets/icons/ic-arrow-backward.svg'
+import { ReactComponent as Error } from '../../../../assets/icons/ic-alert-triangle.svg'
+import { BranchRegexModalProps } from './types'
+
+function BranchRegexModal({
+    material,
+    selectedCIPipeline,
+    showWebhookModal,
+    title,
+    isChangeBranchClicked,
+    context,
+    onClickNextButton,
+    onShowCIModal,
+    handleRegexInputValue,
+    regexValue,
+    onCloseBranchRegexModal,
+}: BranchRegexModalProps) {
+    const getBranchRegexName = (gitMaterialId: number): string => {
+        if (Array.isArray(selectedCIPipeline?.ciMaterial)) {
+            for (let _ciMaterial of selectedCIPipeline.ciMaterial) {
+                if (
+                    _ciMaterial.gitMaterialId === gitMaterialId &&
+                    _ciMaterial.source &&
+                    _ciMaterial.source?.type === SourceTypeMap.BranchRegex
+                ) {
+                    return _ciMaterial.source.regex
+                }
+            }
+        }
+        return ''
+    }
+
+    const renderBranchRegexMaterialHeader = (close: () => void) => {
```

Comment:
rename the close arg to closeModal or something or Close icon to `CloseIcon` something just to avoid confusion.